rust/src/comb/measure_cast/zCDP_to_approxDP/cks20.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2022 OpenDP
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Algorithm from:
+//     Clément Canonne, Gautam Kamath, Thomas Steinke. The Discrete Gaussian for Differential Privacy. 2020.
+//     https://arxiv.org/abs/2004.00010
+//
+// This file is derived from the following implementation by Thomas Steinke:
+//     https://github.com/IBM/discrete-gaussian-differential-privacy/blob/cb190d2a990a78eff6e21159203bc888e095f01b/cdp2adp.py
+
+use crate::{error::Fallible, traits::Float};
+
+pub(crate) fn cdp_epsilon<Q: Float>(rho: Q, delta: Q) -> Fallible<Q> {
+    let tol = Q::round_cast(1e-8)?;
```

Comment:
How do we know this tolerance will always be reached?

Reply:
The tolerance is on epsilon, not delta. The epsilon difference is always halved each iteration. The algorithm always runs a fixed number of steps, based on epsilon and rho. Specifically, `log2(e_max / tol)` steps. If delta is 1e-8 and rho is 1, this is ~30 steps. A few steps less if rho is .05.